core/runtime/src/main/java/io/quarkus/runtime/logging/DecorateStackUtil.java

```diff
@@ -130,6 +130,9 @@ private static List<String> getRelatedLinesInSource(Path filePath, int lineNumbe
 
     private static String getFullPath(String fullClassName, String fileName) {
         int lastDotIndex = fullClassName.lastIndexOf(".");
+        if (lastDotIndex == -1) {
+            return fileName;
+        }
         String packageName = fullClassName.substring(0, lastDotIndex);
         String path = packageName.replace('.', '/');
         return path + "/" + fileName;
```

Comment:
```suggestion
    private static String getFullPath(String fullClassName, String fileName) {
        int lastDotIndex = fullClassName.lastIndexOf(".");
        return (lastDotIndex == -1) 
        ? fileName 
        : fullClassName.substring(0, lastDotIndex).replace('.', '/') + "/" + fileName;
```
as now trinity involved could use dedicated operator to push cohesion.

Reply:
might consider one dedicated test for this fix to impose QA:

```java
import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

class PathUtilsTest {

    @Test
    void getFullPath_WithPackage_ReturnsCorrectPath() {
        String result = PathUtils.getFullPath("com.example.MyClass", "file.txt");
        assertEquals("com/example/file.txt", result);
    }

    @Test
    void getFullPath_DefaultPackage_ReturnsFileNameOnly() {
        String result = PathUtils.getFullPath("MyClass", "file.txt");
        assertEquals("file.txt", result);
    }

    @Test
    void getFullPath_MultiLevelPackage_ReturnsCorrectPath() {
        String result = PathUtils.getFullPath("a.b.c.d.e.F", "file.txt");
        assertEquals("a/b/c/d/e/file.txt", result);
    }

    @Test
    void getFullPath_EmptyClassName_ReturnsFileNameOnly() {
        String result = PathUtils.getFullPath("", "file.txt");
        assertEquals("file.txt", result);
    }

    @Test
    void getFullPath_ClassNameEndsWithDot_ReturnsCorrectPath() {
        String result = PathUtils.getFullPath("com.example.", "file.txt");
        assertEquals("com/example/file.txt", result);
    }

    @Test
    void getFullPath_NullClassName_ReturnsFileNameOnly() {
        String result = PathUtils.getFullPath(null, "file.txt");
        assertEquals("file.txt", result);
    }

    @Test
    void getFullPath_EmptyFileName_ReturnsPathWithEmptyFileName() {
        String result = PathUtils.getFullPath("com.example.MyClass", "");
        assertEquals("com/example/", result);
    }
}
```

Reply:
> as now trinity involved could use dedicated operator to push cohesion.

I would argue this is a more impactful change and less readable overall, but I'm somewhat indifferent to the implementation as long as it solves the problem 😛

Reply:
There is no reason to have a lengthy discussion for this fix.
The original implementation was spot on

Reply:
> might consider one dedicated test for this fix to impose QA:

I agree in principle, but the change is trivial and `DecorateStackUtil` has no tests at the moment. If tests are desired, I suggest the core team would be better positioned to provide more robust coverage.

Reply:
The proposed test is useless

Reply:
this should test it right? 

```
   @Test
    public void testGetFullPath_NoPackage() {
        // Test case where fullClassName has no dots (no package)
        String fullClassName = "SimpleClass";
        String fileName = "testFile.txt";
        
        String result = YourClass.getFullPath(fullClassName, fileName);
        
        // Should return just the fileName since there's no package
        assertEquals(fileName, result);
    }
```
there is actually a negation necessary to have spelling API  !fullClassName.endsWIth("."),

or is it a contains? !fullClassName.contains(".").

one if this must be as its not sure to me what indexOf does and i have to check API doc. When having speaking code its easy to understand. Thats the reason behind. Make code very clear and simple. Considering prose does not contains mathematical terms.